### DISCUSSION TITLE
Index with relative time

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#13092] [Q&A]
Is it possible to produce an index with relative time offsets e.g Uploaded 5minutes ago, Uploaded 1 hr ago, uploaded today, uploaded yesterday

**Comment by mcrio** [comment]
Hi. I guess that would be difficult as what yesterday was `1h ago` is today `1 day ago`. I'd index the date or date-time and handle the relative time client side. ...unless I misunderstood the requirement

**Comment by ayende** [comment]
What is the scenario that you need that for? It looks like a difference between how you display things and how you work with them.
Can you give more details on the scenario?

**Reply by bobskigit** [reply]
In our reporting function, we want to have various counters but some of the counters include items that expire over time. I want to potentially avoid just doing a query again and again to find everything that is older than the given expiry date. 

It's similar to a shopping basket with an inbuilt expiry, if you don't confirm it within x time it removes it from your basket. The business wants to see how many items could potentially convert to a real item

**Reply by ayende** [reply]
Isn't that basically a facet question?

Something like:

Params: `$weekAgo, $3monthsAgo, $6monthsAgo`

And the query being:

```
from index 'Orders/Summary'
where // whatever you filter here
select facet(Date < $6monthsAgo, 
                   Date between $6monthsAgo and $3monthsAgo, 
                   Date between $3monthsAgo and $weekAgo, 
                   Date > $weekAgo)
```

That would give you the total of all of those in a single query.

**Reply by bobskigit** [reply]
Yes, it is, didn't think you could use facet for datetimeoffset types.

Will try it.

thank you
